Literature research (#11)
Dove6/WIMU10: DESIGN_PROPOSAL.md

Dla obu zbiorów wyznaczymy wartości metryk opracowanych przez nas, jak i obecnych już w MusPy (celem porównania).

### Analiza wybranych zbiorów danych dostępnych w MusPy

> W dalszej analizie literatury skupiliśmy się na znajdujących się zbiorach danych w formacie MIDI. Udało nam się zawęzić liczbę potencjalnych kandydatów do pięciu. Ostatecznie posprawdziliśmy czy w istniejących artykułach lub materiałach naukowych nie istniały wzmianki dotyczące problemów napotkanych z danym zestawem danych lub potrzebnym dodatkowym przetwarzaniem wstępnym. Wyniki analizy są następujące:

Reviewer: ```suggestion
Przy analizie zbiorów danych ustaliliśmy następujące ograniczenia:
* zbiory muszą być dostępne poprzez interfejs MusPy,
* interesują nas tylko utwory w formacie MIDI,
* nie uwzględniamy chorałów i hymnów.
W ten sposób otrzymaliśmy listę pięciu zbiorów danych nadających się do naszego zadania, których analizę zamieszczamy poniżej. Zagadnienia, które uznaliśmy za ważne, to:
* jakość plików MIDI/transkrypcji,
* sposób pozyskania danych,
* rozmiar zbioru danych,
* rodzaj muzyki,
* liczba instrumentów/kanałów,
* ewentualna potrzeba dodatkowego przetwarzania,
* popularność/wykorzystanie w innych pracach.
```

- The Lakh MIDI Dataset [^raffel2016dataset] [^raffel2016] - Największy z oferowanych zbiorów danych. Zawiera 176 581 różnych utworów w formacie MIDI. 45 tys. z nich znalazło dopasowanie w wpisach z zestawu danych 'Million Song Dataset' [^bertin-mahieux2011]. 'The Lakh MIDI Dataset' powstał przy pomocy wygenerowanych przez użytkowników plików oraz plików podebranych z internetu. Na wskutek tego w zbiorze istnieje znaczne zaszumienie. Wiele piosenek składa się wyłącznie z kilku nut. Na podstawie tego zestawu danych powstał zestaw 'Lakh Pianoroll Dataset' [^dong2017dataset] [^dong2017], który po oczyszceniu z zbędnego szumu zawiera 174 154 plików. Ten zestaw został później wykorzystany w wytrenowaniu modelu MuseGAN [^dong2017].

> - MAESTRO [^hawthorne2018] - Zestaw zawierający 200 godzin utworów z muzyki klasycznej. Udostępnia zarówno pliki w formacie MIDI, jak również ich odpowiedniki w formacie .wave oraz pliki w formacie .csv i .json zawierające rozpiskę wszystkich utworów, informację o artyście, czas trwania oraz nazwę utworu. Zbiór powstał przy współpracy z organizatorami 'International Piano-e-Competition', gdzie wszystkie utwory były wykonywane na pianinach zawierających zintegrowane systemy przechwytywania i odtwarzania MIDI. W przypadku tego zestawu nie udało się znaleźć żadnych wspomnień o dodatkowym wcześniejszym przetworzeniu. W artykule "A Lightweight Instrument-Agnostic Model for Polyphonic Note Transcription and Multipitch Estimation" [^bittner2022] autorzy nie wspominają o przetwarzaniu wstępnym w przypadku tego zestawu, ponadto podkreślają że korzystają z publicznie dostępnych zestawów danych by zapewnić wysoką potwarzalność wyników, co tylko wspiera ideę że nie wymaga przetwarzania wstępnego by móc z nim pracować. 

Reviewer: ```suggestion
- MAESTRO [^hawthorne2018] - Zestaw zawierający 1282 utworów (~201h) muzyki klasycznej na fortepian. Udostępnia zarówno pliki w formacie MIDI, jak również ich odpowiedniki w formacie WAV oraz metadane w formacie CSV i JSON zawierające informacje o artystach, czasy trwania oraz nazwy utworów. Zbiór powstał przy współpracy z organizatorami konkursu "International Piano-e-Competition", podczas którego wszystkie utwory były wykonywane na fortepianach Yamaha Disklavier nagrywających MIDI. Autorzy nie podają informacji o wcześniejszym przetworzeniu danych, a artykuły wykorzystujące MAESTRO [^bittner2022] nie wskazują na to, że jest one potrzebna do pracy z zestawem.
```

> - NES-MDB (Nintendo Entertainment System Music Database) [^donahue2018] - Zestaw składający się z 5278 różnych piosenek zebranych z 397 gier na NES (Nintendo Entarteimnet System). Głównie wieloinstrumentalne utwory w MIDI. W porównaniu do pozostałych zestawów danych, bardzo mała ilość artykułów powstała z wykorzystaniem tego zestawu.

Reviewer: ```suggestion
- NES-MDB (Nintendo Entertainment System Music Database) [^donahue2018] - Zestaw składający się z 5278 różnych utworów (~46h) zebranych z 397 gier na konsolę NES (Nintendo Entertainment System). Pliki MIDI zostały otrzymane przez konwersję z natywnego dla NES formatu muzyki (instrukcji wbudowanego czterokanałowego syntezatora dźwięku). W porównaniu z pozostałymi zestawami danych, bardzo mała ilość artykułów wykorzystuje NES-MDB.
```

> - MusicNet [^thickstun2017] - Zbiór 330 piosenek z należących do muzyki klasycznej. Zawiera zestawy z plaikami MIDI oraz odpowiadające im zestawy z formatami.wav oraz metadanymi opisującymi piosenki. W skład wchodzą piosenki następujących kompozytorów: Bach, Beethoven, Faure, Brahms, Haydn, Cambini, Dvorak, Mozart oraz Schubert. W artykule "MT3: Multi-Task Multitrack Music Transcription" [^gardner2022] w zestawieniu z kilkoma innymi zestawami danych (m.in. MAESTROv3 [^hawthorne2018], Slakh2100 [^manilow2019], Cerberus4 [^manilow2020], GuitarSet [^xi2018], URMP [^li2018]) zestaw MusicNet wypadł najgorzej. Zestaw powstał dzięki wygenerowanym przez ludzi transkrypcjom. Etykiety były natomiast wyrównywane z nagraniami przy pomocy DWT (Dynamic Time Warping). Na wskutek tego etykiety są w tym zestawie mniej dokładne niż w pozostałych.

Reviewer: ```suggestion
- MusicNet [^thickstun2017] - Zbiór 330 wieloinstrumentalnych utworów muzyki klasycznej (~30h) zawierający pliki MIDI, WAV, CSV (utwory w postaci etykiet nut) i metadane. W skład wchodzą utwory następujących kompozytorów: Bach, Beethoven, Faure, Brahms, Haydn, Cambini, Dvorak, Mozart oraz Schubert. Pliki MIDI to publicznie dostępne, ręcznie wykonane transkrypcje zapisu nutowego. W artykule "MT3: Multi-Task Multitrack Music Transcription" [^gardner2022] w zestawieniu z kilkoma innymi zestawami danych (MAESTROv3 [^hawthorne2018], Slakh2100 [^manilow2019], Cerberus4 [^manilow2020], GuitarSet [^xi2018], URMP [^li2018]) dla zestawu MusicNet osiągnięto najniższe wyniki, wspominając o niewielkim rozmiarze materiału uczącego i złym wyrównaniu w czasie nagrań i etykiet. Druga z wymienionych wad nie ma dla nas znaczenia, ponieważ operujemy wyłącznie na reprezentacji symbolicznej.
```

> - EMOPIA (A Multi-Modal Pop Piano Dataset For Emotion Recognition and Emotion-based Music Generation) [^hung2021] - Zestawienie 1087 klipów z 387 piosenek. Zbiór skupia się na postrzegenaiu emocji w muzyce popowej grywanej na pianinie. Wszystkie utwory na podstawie ich walencji oraz pobudzenia zostały przydzielone do jednej z czterych grup. Zestaw częściej wykorzystywany w pracach naukowych niż NES-MDB. Mimo tego nie udało nam się znaleźć żadnych błędów dotyczących danych lub dodatkowego przetwarzania wstępnego wykorzystywanego podczas pracy z tym zestawem.

Reviewer: ```suggestion
- EMOPIA (A Multi-Modal Pop Piano Dataset For Emotion Recognition and Emotion-based Music Generation) [^hung2021] - Zestawienie 1087 fragmentów 387 utworów (~11h), skupione na postrzeganiu emocji w muzyce popowej/filmowej na pianino. Wszystkie fragmenty zostały przydzielone do jednej z czterech grup na podstawie ich pozytywności ("valence") oraz pobudzenia ("arousal"). Transkrypcję z audio do MIDI otrzymano przy wykorzystaniu modelu AI [^kong2020].
```

Dla obu zbiorów wyznaczymy wartości metryk opracowanych przez nas, jak i obecnych już w MusPy (celem porównania).

### Analiza wybranych zbiorów danych dostępnych w MusPy
Przy analizie zbiorów danych ustaliliśmy następujące ograniczenia:
* zbiory muszą być dostępne poprzez interfejs MusPy,
* interesują nas tylko utwory w formacie MIDI,
* nie uwzględniamy chorałów i hymnów.
W ten sposób otrzymaliśmy listę pięciu zbiorów danych nadających się do naszego zadania, których analizę zamieszczamy poniżej. Zagadnienia, które uznaliśmy za ważne, to:
* jakość plików MIDI/transkrypcji,
* sposób pozyskania danych,
* rozmiar zbioru danych,

> * rodzaj muzyki,

Reviewer 2: W sensie gatunek czy co?

Reviewer 2: ```suggestion
* gatunek muzyki,
```

* liczba instrumentów/kanałów,
* ewentualna potrzeba dodatkowego przetwarzania,
* popularność/wykorzystanie w innych pracach.

> - The Lakh MIDI Dataset [^raffel2016dataset] [^raffel2016] - Największy z dostępnych zbiorów danych. Zawiera 176 581 różnych utworów w formacie MIDI (>5000h), 45 tys. z których znalazło dopasowanie w wpisach z zestawu danych "Million Song Dataset" [^bertin-mahieux2011]. Zbiór zawiera wyłącznie pliki pobrane z Internetu z publicznie dostępnych źródeł. Przy zbieraniu których nie stosowano żadnej kontroli jakości, w konsekwencji znajdują się w nim także pliki o niepoprawnym formacie, wyjątkowo krótkie lub niskiej jakości. Na podstawie tego zestawu danych powstał zestaw "Lakh Pianoroll Dataset" [^dong2017dataset] [^dong2017], który po oczyszczeniu zawiera 174 154 plików. Ten zestaw został później wykorzystany w wytrenowaniu modelu MuseGAN [^dong2017].

Reviewer: s/których/danych/, resztka z poprzedniej wersji zdania.

> - The Lakh MIDI Dataset [^raffel2016dataset] [^raffel2016] - Największy z dostępnych zbiorów danych. Zawiera 176 581 różnych utworów w formacie MIDI (>5000h), 45 tys. z których znalazło dopasowanie w wpisach z zestawu danych "Million Song Dataset" [^bertin-mahieux2011]. Zbiór zawiera wyłącznie pliki pobrane z Internetu z publicznie dostępnych źródeł. Przy zbieraniu których nie stosowano żadnej kontroli jakości, w konsekwencji znajdują się w nim także pliki o niepoprawnym formacie, wyjątkowo krótkie lub niskiej jakości. Na podstawie tego zestawu danych powstał zestaw "Lakh Pianoroll Dataset" [^dong2017dataset] [^dong2017], który po oczyszczeniu zawiera 174 154 plików. Ten zestaw został później wykorzystany w wytrenowaniu modelu MuseGAN [^dong2017].

Reviewer 2: ```suggestion
- The Lakh MIDI Dataset [^raffel2016dataset] [^raffel2016] - Największy z dostępnych zbiorów danych. Zawiera 176 581 różnych utworów w formacie MIDI (>5000h), 45 tys. z których znalazło dopasowanie we wpisach z zestawu danych "Million Song Dataset" [^bertin-mahieux2011]. Zbiór zawiera wyłącznie pliki pobrane z Internetu z publicznie dostępnych źródeł bez zachowania kontroli jakości. W konsekwencji w zbiorze znajdują się także pliki o niepoprawnym formacie, wyjątkowo krótkie lub niskiej jakości. Na podstawie tego zestawu danych powstał zestaw "Lakh Pianoroll Dataset" [^dong2017dataset] [^dong2017], który po oczyszczeniu zawiera 174 154 plików. Ten zestaw został później wykorzystany w wytrenowaniu modelu MuseGAN [^dong2017].
```

> - MAESTRO [^hawthorne2018] - Zestaw zawierający 1282 utworów (~201h) muzyki klasycznej na fortepian. Udostępnia zarówno pliki w formacie MIDI, jak również ich odpowiedniki w formacie WAV oraz metadane w formacie CSV i JSON zawierające informacje o artystach, czasy trwania oraz nazwy utworów. Zbiór powstał przy współpracy z organizatorami konkursu "International Piano-e-Competition", podczas którego wszystkie utwory były wykonywane na fortepianach Yamaha Disklavier nagrywających MIDI. Autorzy nie podają informacji o wcześniejszym przetworzeniu danych, a artykuły wykorzystujące MAESTRO [^bittner2022] nie wskazują na to, że jest one potrzebna do pracy z zestawem.

Reviewer 2: ```suggestion
- MAESTRO [^hawthorne2018] - Zestaw zawierający 1282 utworów (~201h) muzyki klasycznej na fortepian. Udostępnia zarówno pliki w formacie MIDI, jak również ich odpowiedniki w formacie WAV oraz metadane w formacie CSV i JSON zawierające informacje o artystach, czasy trwania oraz nazwy utworów. Zbiór powstał przy współpracy z organizatorami konkursu "International Piano-e-Competition", podczas którego wszystkie utwory były wykonywane na fortepianach Yamaha Disklavier nagrywających MIDI. Autorzy nie podają informacji o procedurach przygotowania danych do użycia, a artykuły wykorzystujące MAESTRO [^bittner2022] nie wskazują, by jakiekolwiek przetwarzanie wstępne było potrzebne do pracy z zestawem.
```

[^thickstun2017]: ["Learning Features of Music from Scratch", John Thickstun et al., 2017](https://arxiv.org/abs/1611.09827)
[^gardner2022]: ["MT3: Multi-Task Multitrack Music Transcription", Josh Gardner et al., 2022](https://arxiv.org/abs/2111.03017v4)
[^hung2021]: ["EMOPIA: A Multi-Modal Pop Piano Dataset For Emotion Recognition and Emotion-based Music Generation", Hsiao-Tzu Hung et al., 2021](https://arxiv.org/abs/2108.01374)
[^bertin-mahieux2011]: ["The Million Song Dataset", Thierry Bertin-Mahieux, 2011](https://academiccommons.columbia.edu/doi/10.7916/D8NZ8J07)
[^manilow2019]: ["Cutting Music Source Separation Some Slakh: A Dataset to Study the Impact of Training Data Quality and Quantity", Ethan Manilow et al., 2019](https://arxiv.org/abs/1909.08494)
[^manilow2020]: ["Simultaneous Separation and Transcription of Mixtures with Multiple Polyphonic and Percussive Instruments", Ethan Manilow et al., 2020](https://arxiv.org/abs/1910.12621)
[^xi2018]: [Guitarset: A Dataset for Guitar Transcription, Qingyang Xi et al., 2018](https://archives.ismir.net/ismir2018/paper/000188.pdf)

> [^li2018]: ["Creating a multi-track classical music performance dataset for multi-modal music analysis: Challenges, insights, and applications", Bochen Li et al., 2018](https://labsites.rochester.edu/air/publications/li2018creating.pdf)

Reviewer: ```suggestion
[^li2018]: ["Creating a multi-track classical music performance dataset for multi-modal music analysis: Challenges, insights, and applications", Bochen Li et al., 2018](https://labsites.rochester.edu/air/publications/li2018creating.pdf)
[^kong2020]: ["High-resolution Piano Transcription with Pedals by Regressing Onset and Offset Times", Qiuqiang Kong et al., 2020](https://arxiv.org/abs/2010.01815)
```
